File: Range Sum Subarrays.py
T = int(input())
for z in range(T):
    N, A, B = map(int, input().split())
    arr = list(map(int, input().split()))
    ans = 0
    for i in range(N):
        for j in range(i, N):
            if A <= sum(arr[i: j + 1]) <= B:
                ans += 1
    print(ans)

File: RangeSumSubArrays.java
package com.vishwa;

import java.util.Scanner;

public class RangeSumSubArrays {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        for (int z = 0; z < t; z++) {
            int N = scanner.nextInt();
            int A = scanner.nextInt();
            int B = scanner.nextInt();
            int ans = 0;
            int[] arr = new int[N];
            for (int i = 0; i < N; i++) {
                arr[i] = scanner.nextInt();
            }
            for (int i = 0; i < N; i++) {
                for (int j = i; j < N; j++) {
                    int s = 0;
                    for (int k = i; k <= j; k++) {
                        s += arr[k];
                    }
                    if((A <= s) && (s <= B)){
                        ans += 1;
                    }
                }
            }
            System.out.println(ans);
        }
    }
}
